templates/prometheus-servicemonitor.yaml

```diff
@@ -41,8 +41,19 @@ spec:
     params:
       format:
       - prometheus
+    {{- $tlsConfig := .Values.serverTelemetry.serviceMonitor.tlsConfig }}
+    {{- if $tlsConfig }}
```

Comment:
Wouldn't that be more lines because I still need the `if` statement but then I need an extra `end` statement as well? e.g.

```yaml
{{- with .Values.serverTelemetry.serviceMonitor.authorization }}
{{- if . }}
authorization:
{{- toYaml . | nindent 6 }}
{{- end }}
{{- end }}
```

Reply:
And for direct comparison, it's currently:

```yaml
{{- $authz := .Values.serverTelemetry.serviceMonitor.authorization }}
{{- if $authz }}
authorization:
{{- toYaml $authz | nindent 6 }}
{{- end }}
```

IMO this version is easier to read.

Reply:
Oh I forgot, `with` generates nothing if the value is empty right? So it would actually be:

```yaml
{{- with .Values.serverTelemetry.serviceMonitor.authorization }}
authorization:
{{- toYaml . | nindent 6 }}
{{- end }}
```